### PR TITLE
[Issue 65] Parsing CSV with '\t' in notebook fails

### LINKED ISSUE #65: Parsing CSV with '\t' in notebook fails
```sql
CREATE TABLE tbl_activity_g0_t60_raw
USING com.databricks.spark.csv
options (path "data", header "false", delimiter '\t')
```
Will fail with:
```
"Delimiter cannot be more than one character."
```

### PR DESCRIPTION
**Pull request by falaki**
We use apache commons-lang to unescape the delimiter string. Also added a unit test for this.

closes #65

### CODE REVIEW

**Comment by rxin** [comment]
Do we really need an extra library for this?

**Comment by falaki** [comment]
Not really, have a better standard way of unescaping strings? I am open to using anything else.

**Comment by brkyvz** [comment]
I don't know a better way of unescaping strings, therefore I'm going to LGTM.

File: src/main/scala/com/databricks/spark/csv/DefaultSource.scala [hunk 1]
```diff
@@ -49,7 +50,7 @@ class DefaultSource
       parameters: Map[String, String],
       schema: StructType) = {
     val path = checkPath(parameters)
-    val delimiter = parameters.getOrElse("delimiter", ",")
+    val delimiter = StringEscapeUtils.unescapeJava(parameters.getOrElse("delimiter", ","))
```
Comment (rxin) [inline src/main/scala/com/databricks/spark/csv/DefaultSource.scala]: can't we just replace "\t" string literal with a tab here? I know there are other things, but I don't think we should handle everything anyway (like \n)

Comment (falaki) [inline src/main/scala/com/databricks/spark/csv/DefaultSource.scala]: I prefer the more general approach. There are some crazy delimiters out there in the wild!

Comment (rxin) [inline src/main/scala/com/databricks/spark/csv/DefaultSource.scala]: What happens if "\n" is specified as the delimiter? Is this library going to crash Spark?

**Pull request merged**